R/clean_curated_data.R

```diff
@@ -5,63 +5,178 @@ fetch_syn_table <- function(syn_id, cols = "*") {
+  }
+  if (has_name(table_df, "Lost_Tracking")) {
+    table_df <- table_df %>% mutate(Lost_Tracking = as.logical(Lost_Tracking))
+  }
+  if (has_name(table_df, "Out_of_Focus")) {
+    table_df <- table_df %>% mutate(Out_of_Focus = as.logical(Out_of_Focus))
+  }
+  if (has_name(table_df, "XCoordinate")) {
+    table_df <- table_df %>% mutate(XCoordinate = as.numeric(XCoordinate))
+  }
+  if (has_name(table_df, "YCoordinate")) {
+    table_df <- table_df %>% mutate(YCoordinate = as.numeric(YCoordinate))
+  }
+  table_df <- table_df %>% select(-ROW_ID, -ROW_VERSION)
   return(as_tibble(table_df))
 }
 
-get_curated_cell_data <- function() {
-  curated_cell_data <- fetch_syn_table("syn11378063") %>%
-    select(-ROW_ID, -ROW_VERSION) %>%
-    mutate(Live_Cells = as.logical(Live_Cells),
-           Lost_Tracking = as.logical(Lost_Tracking))
-  return(curated_cell_data)
+fetch_syn_csv <- function(syn_id) {
+  f <- synGet(syn_id)
+  df <- read_csv(f$path) %>%
+    as_tibble()
+  if (has_name(df, "ObjectTrackID")) {
+    df <- df %>% mutate(ObjectTrackID = as.integer(ObjectTrackID))
+  }
+  if (has_name(df, "TimePoint")) {
+    df <- df %>% mutate(TimePoint = as.integer(TimePoint))
+  }
+  if (has_name(df, "Live_Cells")) {
+    df <- df %>% mutate(Live_Cells = as.logical(Live_Cells))
+  }
+  if (has_name(df, "Mistracked")) {
+    df <- df %>% mutate(Mistracked = as.logical(Mistracked))
+  }
+  if (has_name(df, "Lost_Tracking")) {
+    df <- df %>% mutate(Lost_Tracking = as.logical(Lost_Tracking))
+  }
+  if (has_name(df, "Out_of_Focus")) {
+    df <- df %>% mutate(Out_of_Focus = as.logical(Out_of_Focus))
+  }
+  if (has_name(df, "XCoordinate")) {
+    df <- df %>% mutate(XCoordinate = as.numeric(XCoordinate))
+  }
+  if (has_name(df, "YCoordinate")) {
+    df <- df %>% mutate(YCoordinate = as.numeric(YCoordinate))
+  }
+  return(df)
 }
 
 get_corrected_indices <- function() {
   corrected_abc <- fetch_syn_table("syn17087846")
   corrected_lincs <- fetch_syn_table("syn17096732")
   corrected_sod <- fetch_syn_table("syn17933845")
   corrected_indices <- bind_rows(corrected_abc, corrected_lincs, corrected_sod) %>%
-    select(Experiment, ObjectTrackID, Well, TimePoint, Live_Cells, Lost_Tracking) %>%
+    #select(Experiment, ObjectTrackID, Well, TimePoint, Live_Cells, Lost_Tracking) %>%
     mutate(Live_Cells = as.logical(Live_Cells),
            Lost_Tracking = as.logical(Lost_Tracking))
   return(corrected_indices)
 }
 
-get_potential_errors_corrected <- function() {
-  potential_errors_corrected <- synGet("syn18134075")$path %>%
-    read_csv() %>%
-    as_tibble() %>%
-    select(Experiment, ObjectTrackID, Well, TimePoint, Live_Cells, Lost_Tracking) %>%
-    mutate(ObjectTrackID = as.integer(ObjectTrackID),
-           TimePoint = as.integer(TimePoint))
+fill_in_missing_timepoints <- function(curated_cell_data) {
+  timepoint_reference <- fetch_syn_table(
+    "syn11817859", c("Experiment", "TimePointBegin", "TimePointEnd")) %>%
```

Comment:
pass Syn id as a parameter

Reply:
why is fxn duplicated here and in R/curate_curated_cell_table.R?

Reply:
`R/curate_curated_cell_table.R` is an old version of `R/clean_curated_data.R`. For some reason I decided to nuke `R/curate_curated_cell_table.R` and start over. I'll take it out.